src/compiler/binder.ts

```diff
@@ -144,7 +163,10 @@ module ts {
             let symbol: Symbol;
             if (name !== undefined) {
                 symbol = hasProperty(symbols, name) ? symbols[name] : (symbols[name] = createSymbol(0, name));
-                if (symbol.flags & excludes) {
+                
+                // Check for declarations node cannot be merged with. 
+                // Interfaces declarations cannot be merged with non-ambient class declarations, which isn't encoded in SymbolFlags.

```

Comment:
Could you use the `Instantiated` flag at all? Would this overload its meaning too much? @JsonFreeman